tests/test_transplants.py

```diff
@@ -143,6 +143,110 @@ def test_dryrun_reviewers_warns(client, db, phabdouble, auth0_mock):
     assert response.json["confirmation_token"] is not None
 
 
+def test_dryrun_codefreeze_warn(
+    client,
+    db,
+    phabdouble,
+    auth0_mock,
+    codefreeze_datetime,
+    monkeypatch,
+    request_mocker,
+):
+    product_details = "https://product-details.mozilla.org/1.0/firefox_versions.json"
+    request_mocker.register_uri(
+        "GET",
+        product_details,
+        json={
+            "NEXT_SOFTFREEZE_DATE": "two_days_ago",
+            "NEXT_MERGE_DATE": "tomorrow",
+        },
+    )
+    monkeypatch.setattr("landoapi.transplants.datetime", codefreeze_datetime())
+    mc_repo = Repo(
+        tree="mozilla-conduit",
+        url="https://hg.test/mozilla-conduit",
+        access_group=SCM_CONDUIT,
+        commit_flags=[DONTBUILD],
+        product_details_url=product_details,
+    )
+    mc_mock = MagicMock()
+    mc_mock.return_value = {"mozilla-central": mc_repo}
+    monkeypatch.setattr("landoapi.transplants.get_repos_for_env", mc_mock)
+
+    d1 = phabdouble.diff()
+    r1 = phabdouble.revision(diff=d1, repo=phabdouble.repo())
+    phabdouble.reviewer(
+        r1, phabdouble.user(username="reviewer"), status=ReviewerStatus.ACCEPTED
+    )
+
+    response = client.post(
+        "/transplants/dryrun",
+        json={
+            "landing_path": [
+                {"revision_id": "D{}".format(r1["id"]), "diff_id": d1["id"]}
+            ]
+        },
+        headers=auth0_mock.mock_headers,
+    )
+
+    assert 200 == response.status_code
+    assert "application/json" == response.content_type
+    assert response.json["warnings"]
+    assert response.json["warnings"][0]["id"] == 8
+    assert response.json["confirmation_token"] is not None
```

Comment:
When you write `assert`s you should include a statement describing what the assert is specifically trying to test, ie `assert response.json["warnings"], "warnings should be non-empty for a repo with a code freeze"`. This makes updating/removing/fixing tests easier in the future since you've added the relevant context to the test itself, so another dev can decide if the test needs to be fixed/can be removed etc. Please do this as much as possible when you write tests.

As an example, what is `assert response.json["warnings"][0]["id"] == 8` testing? If I update tests and the number `8` becomes `9` is that a failure? Adding context like `"ID of the returned warning should match the ID for the code freeze warning"` makes this much easier to understand.